### PR TITLE
1154 add loc ids

### PR DESCRIPTION
**Pull request by 0robustus1**
This is a partial pull-request for #1154. It shall implement loc/id support for ontologies and their subsites (e.g. children, symbols, mappings, versions). Additionally it contains loc/id support for ontology-members.

The extended loc/id support (for /ref-style domains) will be provided in a later pull-request.

### CODE REVIEW

File: app/helpers/navigation_helper.rb [hunk 1]
```diff
@@ -137,18 +140,20 @@ def in_subcontroller?(page, current_page)
...
     [
-      ['Projects',         [*resource_chain, :projects]],
-      ['Categories',       [*resource_chain, :categories]],
-      ['Tasks',            [*resource_chain, :tasks]],
-      ['License Models',   [*resource_chain, :license_models]],
-      ['Formality Levels', [*resource_chain, :formality_levels]]
+      ['Projects',         locid_for(resource, :projects), :projects],
+      ['Categories',       locid_for(resource, :categories), :categories],
+      ['Tasks',            locid_for(resource, :tasks), :tasks],
+      ['License Models',   locid_for(resource, :license_models), :license_models],
```
Comment (0robustus1) [inline app/helpers/navigation_helper.rb]: readability?

File: config/routes.rb [hunk 2]
```diff
@@ -1,21 +1,57 @@
...
+  ontology_subsites = %i(
+    mappings symbols children
+    sentences theorems comments
+    metadata ontology_versions graphs
+    projects categories tasks license_models formality_levels
+  )
+
+  ontology_subsites.each do |category|
+    specified_get "/:repository_id/*locid///#{category}" => "#{category}#index",
+      as: :"ontology_iri_#{category}",
```
Comment (0robustus1) [inline config/routes.rb]: Nah, not here.

**Comment by 0robustus1** [comment]
Test on jenkins probably fails because there is no libxml available, can you take care of that @eugenk?

**Comment by eugenk** [comment]
Will do, but not today.

File: config/routes.rb (hunk 2 above)
Comment (eugenk) [inline config/routes.rb]: agree

**Comment by eugenk** [comment]
test this please
(This sequence in a comment triggers a new build in jenkins. This pull request should pass now.)

**Comment by eugenk** [comment]
test this build

**Comment by eugenk** [comment]
with fresh seeds, on http://ontohub.dev/default/partial_order//partial_order///mappings, I get:
```
NoMethodError at /default/partial_order/partial_order/mappings
undefined method `source' for nil:NilClass

block (2 levels) in app/views/mappings/index.html.haml:32
```

**Comment by eugenk** [comment]
...and I see this is a problem introduced by the renaming of models. I'm creating a new branch for that one.

For this branch: :+1:

**Pull request merged**